test/e2e-go/features/transactions/proof_test.go

```diff
@@ -90,40 +92,114 @@ func TestTxnMerkleProof(t *testing.T) {
 	a.NoError(err)
 
 	txid := tx.ID()
+	txidSHA256 := tx.IDSha256() // only used for verification
 	confirmedTx, err := fixture.WaitForConfirmedTxn(status.LastRound+10, baseAcct, txid.String())
 	a.NoError(err)
 
-	proofresp, err := client.TxnProof(txid.String(), confirmedTx.ConfirmedRound)
+	blk, err := client.BookkeepingBlock(confirmedTx.ConfirmedRound)
+	a.NoError(err)
+
+	proofresp, err := client.TxnProof(txid.String(), confirmedTx.ConfirmedRound, crypto.Sha512_256)
 	a.NoError(err)
 
-	hashtype, err := crypto.UnmarshalHashType(proofresp.Hashtype)
+	proofrespSHA256, err := client.TxnProof(txid.String(), confirmedTx.ConfirmedRound, crypto.Sha256)
 	a.NoError(err)
 
-	var proof merklearray.Proof
-	proof.HashFactory = crypto.HashFactory{HashType: hashtype}
-	proof.TreeDepth = uint8(proofresp.Treedepth)
-	a.NotEqual(proof.TreeDepth, 0)
-	proofconcat := []byte(proofresp.Proof)
-	for len(proofconcat) > 0 {
-		var d crypto.Digest
-		copy(d[:], proofconcat)
-		proof.Path = append(proof.Path, d[:])
-		proofconcat = proofconcat[len(d):]
+	generateProof := func(h crypto.HashType, prfRsp generated.ProofResponse) (p merklearray.Proof) {
```

Comment:
why not just use a function?

Reply:
It's only used by this test so why contaminate the namespace? also I think it's a bit easier to reason about since you can read the code in a sequential manner.